ndb/src/google/cloud/ndb/model.py

```diff
@@ -336,15 +338,22 @@ def _entity_to_protobuf(entity):
     data = {}
     for cls in type(entity).mro():
         for prop in cls.__dict__.values():
-            if not isinstance(prop, Property) or prop._name in data:
+            if (
+                not isinstance(prop, Property)
+                or isinstance(prop, ModelKey)
```

Comment:
Fixes bug found while doing this work. __key__ (an instance of ModelKey) was getting into the properties being sent to Datastore, which would cause Datastore to puke.